### PR TITLE
updated gemfile to use hydra-editor to handle uris

### LINKED ISSUE #127: Enable Hydra-Editor to handle URIs
There's similar work in Oregon Digital 2, in @jechols' branch.

### PR DESCRIPTION
**Pull request by luisgreg99**
fixes #127

### CODE REVIEW

File: app/decorators/has_hint_option.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class HasHintOption < SimpleDelegator
```
Comment (straleyb) [inline app/decorators/has_hint_option.rb]: This file is to tell people using the ingest form, which fields are CV's. @terrellt Do you think we should keep this in or wait until later to implement this feature?

File: app/models/generic_file.rb [hunk 2]
```diff
@@ -3,17 +3,17 @@ class GenericFile < ActiveFedora::Base
   include Sufia::GenericFile
 
   property :spatial, predicate: ::RDF::DC.spatial do |index|
-    index.as :stored_searchable, :facetable
+    index.as :stored_searchable, :symbol
```
Comment (straleyb) [inline app/models/generic_file.rb]: I thought this was part of #128?

File: config/initializers/hydra_editor.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+HydraEditor.models = ["GenericFile"]
+HydraEditor::Fields::Generator.factory = ScholarsArchive::Fields::InputFactory.new(
+  HydraEditor::Fields::Factory,
+  DecoratorList.new(HasHintOption, HasURIInputType)
```
Comment (straleyb) [inline config/initializers/hydra_editor.rb]: If we decide to not use the HasHintOption quite yet, we can remove it from here as well.

File: config/routes.rb [hunk 4]
```diff
@@ -10,9 +10,10 @@
     get "/users/sign_out", :to => "sessions#destroy"
   end
 
+ 
   # This must be the very last route in the file because it has a catch-all route for 404 errors.
   # This behavior seems to show up only in production mode.
   root to: 'homepage#index'
-
+  mount HydraEditor::Engine => '/'
```
Comment (straleyb) [inline config/routes.rb]: @luisgreg99 Can you explain why you needed this?

File: app/decorators/has_hint_option.rb (hunk 1 above)
Comment (terrellt) [inline app/decorators/has_hint_option.rb]: Probably later.

File: app/forms/file_edit_form.rb [hunk 5]
```diff
@@ -1,7 +1,16 @@
 # app/forms/file_edit_form.rb
 class FileEditForm < FilePresenter
   include HydraEditor::Form
+  delegate :validators, :to => :model
   include HydraEditor::Form::Permissions
-
+  self.model_class = GenericFile
   self.required_fields = [:title, :creator, :tag, :rights]
+
+  def property_hint(property)
```
Comment (straleyb) [inline app/forms/file_edit_form.rb]: Since we dont need hints quite yet, we could take this method out.

File: app/models/generic_file.rb (hunk 2 above)
Comment (luisgreg99) [inline app/models/generic_file.rb]: Yeah, you are right, I must have added it to see how the index would respond, but I forgot to change it back

Comment (straleyb) [inline app/models/generic_file.rb]: :+1:

File: config/routes.rb (hunk 4 above)
Comment (luisgreg99) [inline config/routes.rb]: Sure, I was reading the readme file at https://github.com/jechols/hydra-editor/blob/master/README.md and it suggested to mount the routes in config/routes.rb. I think we might not need to mount it if we decide not to use the default paths.

Comment (straleyb) [inline config/routes.rb]: Ah sweet. I didn't see that. :+1: Looks good.

**Pull request merged**